backend/src/main/java/com/google/sps/data/Match.java

```diff
@@ -1,19 +1,16 @@
 package com.google.sps.data;
 
-/**
- * A match of two participants. Note: Match includes entire Participants (with unique ID per
- * submitted form) instead of just usernames because a username may be involved in multiple matches
- * over time. This issue can be resolved if we delete matches from the datastore after the users are
- * notified of the match.
- */
+import com.google.common.base.MoreObjects;
+
+/** A match between two participants. */
 public final class Match {
 
-  /** Datastore entity ID */
+  /** Match datastore key ID */
   private final long id;
-  /** First participant */
-  private final Participant firstParticipant;
-  /** Second participant */
-  private final Participant secondParticipant;
+  /** First participant username (datastore key name) */
+  private final String firstParticipantUsername;
+  /** Second participant username (datastore key name) */
```

Comment:
I recommended against storing the full Participant so that we could keep the data models fully separate. Otherwise the code that inserts Match objects into the datastore would have to know how to store and retrieve Participants as well.

Reply:
Less overhead, and you only need the username=key to find the whole participant in the datastore.

Reply:
Ok. If we only ever only need the username part, I have no problems. I was just raising the question because it seems the same amount of storage in terms of storing a pointer to an object (I hope this is the implementation in Java) vs a string. But you got valid points here.